multicluster/apis/multicluster/v1alpha1/resourceimport_types.go

```diff
@@ -28,14 +32,41 @@ type ResourceImportSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of ResourceImport. Edit resourceimport_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// ClusterID specifies the member cluster this resource to import to.
+	// When not specified, import to all member clusters.
+	ClusterID string `json:"clusterID,omitempty"`
+	// Name of imported resource.
+	Name string `json:"name,omitempty"`
+	// Namespace of imported resource.
+	Namespace string `json:"namespace,omitempty"`
+	// Kind of imported resource.
+	Kind string `json:"kind,omitempty"`
+
+	// If imported resource is ServiceImport.
+	ServiceImport mcs.ServiceImport `json:"serviceImport,omitempty"`
+	// If imported resource is EndPoints.
+	Endpoints []v1.EndpointSubset `json:"endpoints,omitempty"`
+	// If imported resource is ExternalEntity.
+	ExternalEntity *v1alpha2.ExternalEntitySpec `json:"externalentity,omitempty"`
+	// If imported resource is Node (IPs)
+	Node v1.NodeStatus `json:"node,omitempty"`
+	// If imported resource is ANP.
+	// TODO:
+	// ANP uses float64 as priority.  Type float64 is discouraged by k8s, and is not supported by controller-gen tools.
+	// NetworkPolicy *v1alpha1.NetworkPolicySpec `json:"networkpolicy,omitempty"`
+	// If imported resource Kind is unknown.
+	Raw []byte `json:"raw,omitempty"`
 }
 
 // ResourceImportStatus defines the observed state of ResourceImport
 type ResourceImportStatus struct {
```

Comment:
Should this be an array in ResourceImport in line 81, because each member cluster would write its own status ?

Reply:
K8s norm is that each Resource contains singleton Spec, and potentially singleton Status. Each Spec/Status may contain array of element or single element.

Reply:
So should the below fields be an array? Im wondering how can each member cluster write its status about the import of a resource?

Reply:
you totally right, I skipped one level.

Reply:
done.